lms/resources/_js_config/__init__.py

```diff
@@ -307,21 +307,15 @@ def _add_canvas_speedgrader_settings(self, **kwargs):
             "lis_outcome_service_url"
         )
 
-        # Don't set the Canvas submission params in non-Canvas LMS's.
-        if not self._context.is_canvas:
-            return
-
-        # When a Canvas assignment is launched by a teacher or other
-        # non-gradeable user there's no lis_result_sourcedid in the LTI
-        # launch params.
-        # Don't post submission to Canvas for these cases.
-        if not lis_result_sourcedid:
-            return
-
-        # When a Canvas assignment isn't gradeable there's no
-        # lis_outcome_service_url.
-        # Don't post submission to Canvas for these cases.
-        if not lis_outcome_service_url:
+        if (
+            # Don't set the Canvas submission params in non-Canvas LMS's.
+            not self._context.is_canvas
+            # Don't set the Canvas submission for non-gradable users
+            or not self._lti_user.is_instructor
+            # We need lis_result_sourcedid to talk to the API.
+            # Also on LTI1.1 lis_result_sourcedid == None for non-gradeable users
+            or not lis_result_sourcedid
```

Comment:
There's a change of behavior here. It could that canvas is sometimes sending us the `lis_result_sourcedid` for cases where "is_instructor" would be false.

If that's the case we'll stop sending submissions there but I think is either a bug (ours? canvas?) or something very rare.